dashboard-widgets.el

```diff
@@ -254,15 +265,30 @@ Return entire list if `END' is omitted."
     (cl-subseq seq start (and (number-or-marker-p end)
                               (min len end)))))
 
+(defun dashboard-get-shortcut (item)
+  "Get the shortcut to be used for ITEM."
+  (let ((elem (assq item dashboard-item-shortcuts)))
+    (and elem (cdr elem))))
+
 (defmacro dashboard-insert-shortcut (shortcut-char
                                      search-label
                                      &optional no-next-line)
   "Insert a shortcut SHORTCUT-CHAR for a given SEARCH-LABEL.
 Optionally, provide NO-NEXT-LINE to move the cursor forward a line."
   `(progn
      (eval-when-compile (defvar dashboard-mode-map))
-     (let ((sym (make-symbol (format "Jump to \"%s\"" ,search-label))))
+     (let ((sym (make-symbol (format "dashboard-jump-to-%s" (downcase
+                                                             (replace-regexp-in-string
+                                                              " +" "-" (replace-regexp-in-string
+                                                                        ":+" "" ,search-label
+                                                                        nil nil nil)
+                                                              nil nil nil))))))
        (fset sym (lambda ()
```

Comment:
This function definition happens within the `let` during runtime—it is thrown away after the runtime `let` exits. This is why `C-h f dashboard-jump-to-recent-files` fails. This needs to be replaced with a `defun`, like:

```elisp
  (let ((sym (make-symbol (format "dashboard-jump-to-%s" (downcase search-label)))))
    `(progn
       (eval-when-compile (defvar dashboard-mode-map))
       (defun ,sym nil
         ,(concat
           "Jump to "
           (replace-regexp-in-string
            ":+" "" (format "%s" search-label) nil nil nil)
           ".  This code is dynamically generated in `dashboard-insert-shortcut'.")
         (interactive)
         (unless (search-forward ,search-label (point-max) t)
           (search-backward ,search-label (point-min) t))
         ,@(unless no-next-line
             '((forward-line 1)))
         (back-to-indentation))
```

Reply:
Thanks for the suggestion! I tried this out in another branch, but was not able to get it working. The code works, but indeed I am not able to do either `M-x dashboard-jump-to-projects RET` or `C-h f dashboard-jump-to-project RET` (with the same error as above).

https://github.com/winny-/emacs-dashboard/commit/0e0a56da0b19fbb00a8618119d3ed9d604b0de0c

Reply:
Looking at your linked commit, the problem is that you're using `make-symbol`, which gives you an uninterned symbol, ie. one that is unique. You need to use `intern` instead.

```elisp
(eq 'testing (intern "testing")) ; => t
(eq 'testing (make-symbol "testing")) ; => nil
```

Reply:
Thank you very much for the help! That did the trick! =)